modules/mobile/app/helpers/mobile/list_filter.rb

```diff
@@ -2,54 +2,131 @@
 
 module Mobile
   class ListFilter
-    def initialize(collection, filters)
+    include SentryLogging
+
+    class FilterError < StandardError
+    end
+
+    PERMITTED_OPERATIONS = %w[eq notEq].freeze
+
+    def initialize(collection, filter_params)
       @collection = collection
-      @filters = filters
+      @filter_params = filter_params
     end
 
     # Accepts params:
     #   @collection - a Common::Collection of Common::Base models
-    #   @filters - should be an ActionController::Parameters object which should be passed in from the
+    #   @filter_params - should be an ActionController::Parameters object which should be passed in from the
     #     controller via @params[:filter]. This will pass in another ActionController::Parameters object.
     # Returns: a new Common::Collection of Common::Base models that match the provided filters
-    def self.matches(collection, filters)
-      filterer = new(collection, filters)
+    def self.matches(collection, filter_params)
+      filterer = new(collection, filter_params)
       filterer.result
     end
 
-    # carrying over errors and metadata while merging filters into metadata because
-    # that's how the Common::Collection filter works. these values are probably not used again
     def result
-      metadata = @collection.metadata.merge(filter: @filters)
+      validate!
+      metadata = @collection.metadata.merge(filter: filters)
       Common::Collection.new(data: matches, metadata: metadata, errors: @collection.errors)
+    rescue FilterError => e
+      @collection.errors[:filter_error] = e.message if valid_collection?
+      log_exception_to_sentry(e, extra_context_for_errors)
+      @collection
+    rescue => e
+      @collection.errors[:filter_error] = 'unknown filter error'
+      log_exception_to_sentry(e, extra_context_for_errors)
+      @collection
+    end
+
+    # not adding full collection to extra context because it could be a large amount of data,
```

Comment:
I added the filters and model classes to the sentry logging to give us more breadcrumbs but I think the cost/benefit with adding collections is probably bad.